sql/core/src/main/scala/org/apache/spark/sql/execution/joins/HashedRelation.scala

```diff
@@ -29,16 +31,39 @@ import org.apache.spark.util.collection.CompactBuffer
  */
 private[joins] sealed trait HashedRelation {
   def get(key: Row): CompactBuffer[Row]
+
+  def writeBytes(out: ObjectOutput, serialized: Array[Byte]): Unit = {
```

Comment:
make protected and add a comment that these are helper functions to implement externalizable